frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java

```diff
@@ -57,7 +63,29 @@ public IEnhancedLambdaLogger log() {
   @UIEffect
   public void onUpdate(AnActionEvent anActionEvent) {
     super.onUpdate(anActionEvent);
+
     syncToRemoteStatusDependentActionUpdate(anActionEvent);
+
+    var branchName = getNameOfBranchUnderActionWithLogging(anActionEvent);
+    var relation = branchName.flatMap(bn -> getGitMacheteBranchByNameWithLogging(anActionEvent, bn))
+        .map(b -> b.getSyncToRemoteStatus())
+        .map(strs -> strs.getRelation());
+    var project = tryGetProject(anActionEvent);
+
+    if (branchName.isDefined() && relation.isDefined() && isForcePushRequired(relation.get()) && project.isDefined()) {
+      var anyPatternMatch = GitSharedSettings.getInstance(project.get()).getForcePushProhibitedPatterns().stream()
```

Comment:
hold your horses :horse: :horse: why does one not simply `git4idea.config.GitSharedSettings#isBranchProtected`?

Reply:
Wooooow, how can I miss that method :thinking: Thanks!